### ISSUE TITLE
Loading the "Add Thing" screen is slow

### ISSUE THREAD

**Issue by jimtng** [openhab/openhab-webui#2337]
Clicking on Things -> + leads to the "Choose Binding" page. This page can be quite slow to load, randomly. Usually the first load is slow, then subsequent loads are faster albeit still with a small delay. Then randomly, it would become slow again. This makes for a frustrating experience when trying to add multiple Things.

- Speed up the first load
- Make the subsequent load instantaneous

Could it cache the list of bindings, and show the cache upon first load, and then refresh the list of binding after the page is loaded.
I'm also thinking perhaps once user is on the "Things List" page, start loading the cache of list of binding, so when user clicks Add, it's already loaded / loading.

**Comment by ghys** [comment]
I've experienced the same and think this is heavily linked to whether you have the Z-Wave binding installed or not. Caching is possible but the cache would be invalidated frequently anyways, so it wouldn't help much (or if done badly, would present outdated info).

**Comment by JustinGeorgi** [comment]
> this is heavily linked to whether you have the Z-Wave binding installed or not

I second this theory.  I recently got rid of the z-wave binding and I have noticed an improved performance of the `Add Thing` page.

**Comment by jimtng** [comment]
I don't have zwave

**Comment by jimtng** [comment]
Anyone knows why querying the list of add-ons can be so slow?

**Comment by florian-h05** [comment]
> Anyone knows why querying the list of add-ons can be so slow?

Let's ask the @openhab/core-maintainers maintainers.

**Comment by kaikreuzer** [comment]
I don't. Maybe @J-N-K has any idea?

**Comment by J-N-K** [comment]
I think this maybe related to the marketplace add-on services (community and JSON). They cache the remotely available add-ons for 15 minutes, then refresh on next request. This may result in a slow response when one (or both) of the remote services are slow (or the connection speed of the openHAB instance is low). This also explains why subsequent requests are faster, because the cache is used in the add-on services. It doesn't explain why `zwave` should have an effect.

**Comment by jsjames** [comment]
@florian-h05 - given this is not a ui issue, I recommend we close this one out - maybe with a 'won't fix' label.

**Comment by florian-h05** [comment]
Agreed to close this, in any case the UI cannot do much if the REST API is slow to respond.
